### DISCUSSION TITLE
gtk4-classic

### DISCUSSION THREAD

**Discussion by luigifab** [lah7/gtk3-classic#63] [General Chat]
Even if GTK 4 is still experimental (until 4.20 or 4.24), do you think to port some patches for GTK 4?

**Comment by lah7** [comment]
Sorry! Personally I have **no** incentive or plans to do so. 🙅🏼‍♂️ Any one is welcome to pick up the patches and try them with GTK+4. Maybe the codebases are similar?

GTK+3 should still stick around for many more years. The 🎨 GIMP program (ironically, which started GTK) is still working on their transition from GTK+2 after all!

**Edit:** Related issue: https://github.com/lah7/gtk3-classic/issues/100

**Reply by StarterX4** [reply]
> Any one is welcome to pick up the patches and try them with GTK+4. Maybe the codebases are similar?

I tried to build GTK4.7.1 with the patches untouched – `csd__disabled-by-default.patch` and `other__remove_dead_keys_underline.patch` applied fine. `file-chooser__places-sidebar.patch` applied mostly, and the rest failed.

**Reply by luigifab** [reply]
I tried to build GTK 4.12.3 with [few patches](https://gist.github.com/luigifab/0fce786cdb93b5687069a82f490ea95e) with [openSUSE build farm for Debian Testing](https://build.opensuse.org/package/show/home:luigifab/gtk4-classic):
The deprecated GtkTreeview is so beautiful (pango 1.51) !!

![image](https://github.com/lah7/gtk3-classic/assets/31816829/a3b4c7dd-2f6e-43b6-821f-528a65caa5ff)
`GTK 2                                          GTK 3                                         GTK 4`

**Reply by luigifab** [reply]
If someone know why, with GTK 4:
- when I hover with my mouse a menu bar, the `:selected` state is enabled for my `menubar item`, it's not acceptable because I have a background color, and I want a background color only when I click on the item of the menu bar
- does anyone know the css secret for radio and check in treeview? I tried `treeview.view.radio` and `treeview.view.check` but no...

**Reply by Gamer95875** [reply]
your message is over 2 years old so you've probably already figured out how but 
>     * when I hover with my mouse a menu bar, the `:selected` state is enabled for my `menubar item`, it's not acceptable because I have a background color, and I want a background color only when I click on the item of the menu bar

i used `menubar > item:selected:not(:focus)` for that situation in my theme, i think it worked differently on X11 tho, but it worked fine on Wayland (additionally, that and ` menubar > item:hover` both ended up being really useful for having hovered and pressed menubar items look different, if only that was possible on GTK3 tho)

**Reply by luigifab** [reply]
I had found: `menubar:focus-within item:selected { }` and `menubar:focus-within item:selected:focus, menubar:focus-within item:selected:focus:hover { }`

**Comment by sc0w** [comment]
btw, I am working some years ago in CTK (gtk3 fork), and it just works:

https://github.com/cafe-desktop/ctk

I ported some applications in https://github.com/cafe-desktop (MATE fork) and it works like a charm.

**Comment by MorsMortium** [comment]
I have mentioned this elsewhere, but I have created this to remove CSD in both GTK3 and GTK4: https://codeberg.org/MorsMortium/GTK-NoCSD

**Reply by bayazidbh** [reply]
I currently use this. I've set it in my user environment file, and so far it works well - even with more involved apps like Bottles.

Granted, in such cases, the vertical screen saving isn't THAT much, but at the same time, having SSD is still worth it and it's still a screen space saving that's always useful for my smaller devices.
